### PR TITLE
Update sensor docs

### PR DESCRIPTION
**Pull request by hamishwillee**

### CODE REVIEW

File: en/SetupView/Sensors.md [hunk 1]
```diff
@@ -1,42 +1,118 @@
...
+Click on the button for each sensor to start its calibration sequence.
+
+<span id="flight_controller_orientation"></span>
+## Flight Controller Orientation (Sensor Settings / Set Orientations)
+
+> **Tip** You can skip this section if the flight controller and compass are mounted in the default orientation (upright on the vehicle and facing the front - `ROTATION_NONE`).
+
+If the autopilot/compass are mounted in any other way you will need to specify the orientations using either the **Sensor Settings** or **Set Orientation** buttons. In either case, choose the orientation for your autopilot/sensors from the selection list(s) and then press **OK**.
+
+<img src="../../images/setup/sensor_orientation_sensor_settings.jpg" style="width: 600px;"/>
```
Comment (hamishwillee) [inline en/SetupView/Sensors.md]: @DonLakeFlyer 

- What decides which option you get - Sensor Settings or Set Orientation? 
- If you don't get option to set  (external) compass orientation (Sensor settings) how are those set?

File: en/SetupView/Sensors.md [hunk 2]
```diff
@@ -1,42 +1,118 @@
...
+> <img src="../../images/setup/flight_controller_orientation.png" style="width: 600px;"/>
+>
+> For example, the image below shows some `ROTATION_YAW_XX` orientations:
+> <img src="../../images/setup/flight_controller_yaw_rotation.png" style="width: 600px;"/>
 
-For ArduPilot follow text textual instructions in the center display, clicking the Next button to capture each position.
 
-For PX4 you will be guided with a visual display of positions.
 
 ## Compass
```
Comment (hamishwillee) [inline en/SetupView/Sensors.md]: FYI I've removed the "newer/older firmware" from the headings - this now appears in the body of the docs.

File: en/SetupView/Sensors.md [hunk 3]
```diff
@@ -1,42 +1,118 @@
...
+   
+1. Repeat the calibration process for all vehicle orientations.
+   
+Once you've rotated the vehicle in all the positions *QGroundControl* will display *Calibration complete* (all orientation images will be displayed in green and the progress bar will fill completely). You can then proceed to the next sensor. 
+
+
+### ArduPilot Compass Calibration
 
-![](../../images/setup/ArudPilolCompassCalOnboardResult.jpg)
+ArduPilot uses onboard calibration support that allows for more accurate calibration.
```
Comment (hamishwillee) [inline en/SetupView/Sensors.md]: I would have liked to verify/expand on this but my hardware doesn't work. Could this be tested using a MockLink?

File: en/SetupView/Sensors.md [hunk 4]
```diff
@@ -1,42 +1,118 @@
...
+
+  * For planes this is the position during level flight (planes tend to have their wings slightly pitched up!)
+  * For copters this is the hover position.
+
+1. Click the **Level Horizon** button.
+
+
+## Airspeed
+
+For Airspeed calibration you will need to keep your airspeed sensor out of any wind and then blow across the sensor.
```
Comment (hamishwillee) [inline en/SetupView/Sensors.md]: This is from QGC. Is there anything else to this? (no airspeed sensor to test with). 

I am assuming you press OK to start the calibration. Get asked to blow across the sensor, then after a few seconds you get told calibration is complete. Correct?

File: en/SetupView/Sensors.md [hunk 5]
```diff
@@ -1,42 +1,118 @@
...
+
+To calibrate the airspeed sensor:
+
+1. Click the **Airspeed** sensor button
+1. Click **OK** to start the calibration. 
+
+![Airspeed calibration](../../images/setup/sensors_airspeed.jpg)
+
+
+## CompassMot (Optional - ArduPilot only)
```
Comment (hamishwillee) [inline en/SetupView/Sensors.md]: Also can't verify this. Have made it more clear that this is optional.

**Comment by hamishwillee** [comment]
@DonLakeFlyer I've updated sensor docs here - testing all the PX4 sensors except airspeed. Tested some of ardupilot (that I could). 

Structurally main change is that orientation check is the first item. The reason for this is that a) it means I don't need to explain the orientation in the compass or accelerometer sections, b) it can immediately be skipped if the board is oriented as per usual. 

Appreciate your feedback (there are a couple of questions inline)

File: en/SetupView/Sensors.md [hunk 6]
```diff
@@ -1,42 +1,182 @@
...
+
+If the autopilot/compass are mounted in any other way you will need to specify their orientations as YAW, PITCH and/or ROLL offsets relative to the forward-facing-upright orientation (clock-wise rotation around the Z, Y and X axis, respectively).
+
+<img src="../../images/setup/flight_controller_orientation.png" style="width: 600px;"/>
+
+For example, the image below are at orientations: `ROTATION_NONE`, `ROTATION_YAW_90`,`ROTATION_YAW_180`,`ROTATION_YAW_270`.
+
+<img src="../../images/setup/flight_controller_yaw_rotation.png" style="width: 600px;"/>
+
+### Set Orientations (PX4)
```
Comment (hamishwillee) [inline en/SetupView/Sensors.md]: Why do ArduPilot and PX4 have different menu item text to access same functionality (setting orientations)

File: en/SetupView/Sensors.md (hunk 4 above)
Comment (DonLakeFlyer) [inline en/SetupView/Sensors.md]: Right. The firmware send status text messages for instructions which are shown to the user.

**Comment by DonLakeFlyer** [comment]
This will take me a little bit to review. I want to run through actual calibrations and match against docs.

**Comment by hamishwillee** [comment]
That's great! I ran the calibrations for PX4 (except airspeed) so I am confident of those docs. Couldn't get my iris to work with ardupilot for accel calibration (always failed) which is why that didn't get updated.

From a docs approach view I've tried to minimise the number of diagrams to "just what is needed".

**Comment by DonLakeFlyer** [comment]
I'll do run through with ArduPIlot then.

**Comment by DonLakeFlyer** [comment]
*One thing is for places which show you compass settings, you can have up to three compasses.
* Level Horizon for ArduPilot is slightly different. No option for orientation change.
* Also I've been told that I'm missing some variant of AirSpeed cal for ArdiPilot. I'm going to add that for 3.2. Not sure exactly what that is yet.

**Comment by hamishwillee** [comment]
So can you also see multiple compasses in PX4? If so, can you generate screenshot? Ditto for ArduPilot.

Also Level horizon for Ardupilot?

**Comment by hamishwillee** [comment]
PS None of those things should IMO block this being merged. Right?

**Comment by DonLakeFlyer** [comment]
Multiple compasses show up in PX4 as well. No reason to block merge?

**Comment by hamishwillee** [comment]
> No reason to block merge?

What I am saying is that if I merge this now it is not "wrong", just not necessarily as complete as it would be with additional images I have asked if you can create for me. So the question is - can I merge this now and update with new images later.

Very happy to wait, but it depends on whether you are likely to have the images in a timely manner.

**Comment by DonLakeFlyer** [comment]
Merge away

**Pull request merged**